### DISCUSSION TITLE
My `<ESC>` will just back to normal mode when in lazygit

### DISCUSSION THREAD

**Discussion by LintaoAmons** [folke/snacks.nvim#72] [Q&A] [answered]
I can't use `<ESC>` to quit an action...

How do you guys deal with this situation?

**Comment by folke** [comment]
What do you mean? Your message is not really clear.

**Comment by LintaoAmons** [comment]
![lazygit](https://github.com/user-attachments/assets/80311daf-6f8b-4fee-9d40-2e58948e6771)
hahah, my bad english.
You can see from the gif that I want to abort this `Stash` action, but I can't. My `<ESC>` is acting like exiting the insert mode

**Comment by p10** [comment]
Same problem. Is there a way to send ESC to lazygit?
ESC is a widely used shortcut in LazyGit, but when lazygit is used in the terminal inside nvim, it switches terminal mode - from insert to normal.

**Reply by folke** [reply, marked as answer]
you must do something weird in your config.
In stock snacks terminal, esc in lazygit works as it should

**Reply by p10** [reply]
you're right. i got it mapped: `vim.keymap.set('t', '<esc>', [[<C-\><C-n>]], opts)`.
BTW, you do wonderful job around neovim, i send you my best regards

**Reply by LintaoAmons** [reply]
I found my config as well, I mapped `<ESC>` to return to normal mode in neovim terminal

**Reply by Bobstyle23** [reply]
i was encountering the same problem also
